tool/clean/clean_host/clean_host.go

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+
+package main
+
+import (
+	"context"
+	"github.com/aws/amazon-cloudwatch-agent/tool/clean"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go/aws"
+	"log"
+	"time"
+)
+
+// Clean integration hosts if they have been open longer than 1 day
+func main() {
+	err := cleanHost()
+	if err != nil {
+		log.Fatalf("errors cleaning %v", err)
+	}
+}
+
+func cleanHost() error {
+	log.Print("Begin to clean EC2 Host")
+
+	cxt := context.Background()
+	defaultConfig, err := config.LoadDefaultConfig(cxt)
+	if err != nil {
+		return err
+	}
+	ec2client := ec2.NewFromConfig(defaultConfig)
+
+	terminateInstances(cxt, ec2client)
+	return err
+}
+
+func terminateInstances(cxt context.Context, ec2client *ec2.Client) {
+	maxResults := int32(1000)
+	const tagName = "tag:Name"
+	var tagValue = []string{
+		"buildLinuxPackage",
+		"buildPKG",
+		"buildMSI",
+		"MSIUpgrade_*",
+		"IntegrationTestBase",
+		"CWADockerImageBuilderX86",
+		"CWADockerImageBuilderARM64",
+	}
+	nameFilter := types.Filter{Name: aws.String(tagName), Values: tagValue}
+
+	nextToken := aws.String("")
+	instanceInput := ec2.DescribeInstancesInput{
+		Filters: []types.Filter{
+			nameFilter,
+			{Name: aws.String("instance-state-name"),
+				Values: []string{"running"}}},
+		MaxResults: maxResults}
+	for {
+		instanceIds := make([]string, 0)
+		expirationDateInstance := time.Now().UTC().Add(clean.KeepDurationOneDay)
```

Comment:
Nice ! I like the idea of since we are waiting some time to terminate the previous patch, we can calculate again the dates to terminate the current patch.